src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -229,7 +229,9 @@ public function prepare(Request $request): static
                         $path = $location.substr($path, \strlen($pathPrefix));
                         // Only set X-Accel-Redirect header if a valid URI can be produced
                         // as nginx does not serve arbitrary file paths.
-                        $this->headers->set($type, $path);
+                        // We need to encode the path because nginx fail with 404 if certain
+                        // characters are present in the path.
```

Comment:
no need for the comment IMHO because it's telling about a specific behavior, aka a symptom
this should be encoded because that's the protocol, period :)
```suggestion
```